pkg/envoy/admin/tls/pki.go

```diff
@@ -73,10 +73,14 @@ func GenerateClientCert(ca tls.Certificate) (util_tls.KeyPair, error) {
 	return util_tls.NewCert(*rootCert, ca.PrivateKey.(*rsa.PrivateKey), ClientCertSAN, util_tls.ClientCertType, util_tls.DefaultKeyType, ClientCertSAN)
 }
 
-func GenerateServerCert(ca tls.Certificate, ip string) (util_tls.KeyPair, error) {
+func GenerateServerCert(ca tls.Certificate, address string, advertisedAddress string) (util_tls.KeyPair, error) {
 	rootCert, err := x509.ParseCertificate(ca.Certificate[0])
 	if err != nil {
 		return util_tls.KeyPair{}, err
 	}
-	return util_tls.NewCert(*rootCert, ca.PrivateKey.(*rsa.PrivateKey), ip, util_tls.ServerCertType, util_tls.DefaultKeyType, ip)
+	hosts := []string{address}
+	if advertisedAddress != "" {
+		hosts = append(hosts, advertisedAddress)
+	}
+	return util_tls.NewCert(*rootCert, ca.PrivateKey.(*rsa.PrivateKey), address, util_tls.ServerCertType, util_tls.DefaultKeyType, hosts...)
```

Comment:
Shall the advertisedAddress be used instead of address as third parameter when it is defined?